Restructure Source Maps Upload page (#5799)
getsentry/sentry-docs: src/platform-includes/sourcemaps/overview/javascript.mdx

## Uploading Source Maps to Sentry

We've compiled a list of guides on how to upload source maps to Sentry for the most popular JavaScript build tools:

- <PlatformLink to="/sourcemaps/uploading/webpack/">Webpack</PlatformLink>
- <PlatformLink to="/sourcemaps/uploading/typescript/">TypeScript (tsc)</PlatformLink>
- <PlatformLink to="/sourcemaps/uploading/rollup/">Rollup</PlatformLink>
- <PlatformLink to="/sourcemaps/uploading/vite/">Vite</PlatformLink>
- <PlatformLink to="/sourcemaps/uploading/esbuild/">esbuild</PlatformLink>
- <PlatformLink to="/sourcemaps/uploading/uglifyjs/">UglifyJS</PlatformLink>
- <PlatformLink to="/sourcemaps/uploading/systemjs/">SystemJS</PlatformLink>

### Other Tools

> If you're not using one of these tools, we assume you already know how to generate source maps for your toolchain and we recommend you upload them using the <PlatformLink to="/sourcemaps/uploading/cli/">Sentry CLI</PlatformLink>.

Reviewer: ```suggestion
If you're not using one of these tools, we assume you already know how to generate source maps with your toolchain and we recommend you upload them using <PlatformLink to="/sourcemaps/uploading/cli/">Sentry CLI</PlatformLink>.
```

<PlatformSection notSupported={["node"]}>

<Note>

> Though we strongly recommend uploading source maps as part of your build process, for browser applications it's also possible to <PlatformLink to="/sourcemaps/uploading/hosting-publicly/">host your source maps publicly</PlatformLink>.

Reviewer: I wonder if we should explain why hosting is a bad idea (unless we do that somewhere else).

Author: I don't even know why we don't like this. Gonna leave it for now.

Reviewer: Yup, point 3 is the big one, with the addition that if they do use hashes, they have to host all the old versions of the file/sourcemap forever.
